shell/imports/server/accounts/ldap.js

```diff
@@ -111,22 +109,25 @@ LDAP.prototype.ldapCheck = function (db, options) {
 
     if (_this.options.searchBindDn) {
       let ldapBindFut = new Future();
-      client.bind(_this.options.searchBindDn, _this.options.searchBindPassword,
-        function (err) {
-          if (err) {
-            ldapBindFut.throw(err);
-          } else {
-            ldapBindFut.return();
-          }
+      let searchBindErrorFunc = function (err) {
+        if (err) {
+          ldapBindFut.throw(err);
+        } else {
+          ldapBindFut.return();
         }
-      );
+      };
+
+      client.on("error", searchBindErrorFunc);
+      client.bind(_this.options.searchBindDn, _this.options.searchBindPassword, searchBindErrorFunc);
 
       try {
         ldapBindFut.wait();
       } catch (err) {
         return { error: err };
       }
     }
+
+    client.on("error", errorFunc);
```

Comment:
Is it intended that we may end up registering multiple on-error handlers? What is the effect of running both? You moved this line down here, so I guess ordering is important? Might be worth a comment explaining that this is expected, otherwise it looks like it could be a mistake.